ratis-server-api/src/main/java/org/apache/ratis/server/leader/LeaderState.java

```diff
@@ -47,6 +48,10 @@ public String toString() {
   AppendEntriesRequestProto newAppendEntriesRequestProto(FollowerInfo follower,
       List<LogEntryProto> entries, TermIndex previous, long callId);
 
+  /** obtain current read index by broadcasting heartbeats and maintain authority
+   * @return readIndex, or exception if anything wrong */
+  CompletableFuture<Long> getReadIndex();
```

Comment:
Remove this method from LeaderState.  Declare it in LeaderStateImpl only.

Reply:
done